cloudflare/resource_cloudflare_logpush_job_test.go

```diff
@@ -0,0 +1,73 @@
+package cloudflare
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"os"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccCloudflareLogpushJob_Basic(t *testing.T) {
+	zone := os.Getenv("CLOUDFLARE_DOMAIN")
+	rnd := acctest.RandString(10)
+	name := "cloudflare_logpush_job." + rnd
+	ownershipToken := os.Getenv("CLOUDFLARE_LOGPUSH_OWNERSHIP_TOKEN")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareLogpushJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareLogpushJobConfig(zone, rnd, ownershipToken),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "name", "test-logpush-job"),
+					resource.TestCheckResourceAttr(name, "enabled", "true"),
+					resource.TestCheckResourceAttr(name, "logpull_options", "fields=RayID,ClientIP,EdgeStartTimestamp&timestamps=rfc3339"),
+					resource.TestCheckResourceAttr(name, "destination_conf", "s3://logpush-test-bucket?region=us-west-1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudflareLogpushJobDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudflare.API)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudflare_logpush_job" {
+			continue
+		}
+
+		primaryID, err := strconv.Atoi(rs.Primary.ID)
+
+		if err != nil {
+			return fmt.Errorf("Could not retrieve LogpushJob ID")
+		}
+
+		_, fetchErr := client.LogpushJob(rs.Primary.Attributes["zone_id"], primaryID)
+		if fetchErr == nil {
+			return fmt.Errorf("Logpush job still exists: %s", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudflareLogpushJobConfig(zoneName, ID, ownershipToken string) string {
```

Comment:
This test should now be looking at `zoneID` instead per @patryk's request to update to IDs instead of names.

Reply:
Fixed in 5b2e5d0. 

I also added back this line https://github.com/terraform-providers/terraform-provider-cloudflare/pull/287/commits/5b2e5d00fa24494b7bb1b06f25c988d93d58566e#diff-26590b83973940da3ec408a76663eae3R91 after lots of experimentation with various schema behaviors. 

The issue I'm encountering is that ownership_token is a required field that must be passed when creating or updating a Logpush job. However, it is technically a secret / auth token and is not part of the job definition that will ever be returned by the Logpush API. 

With these changes, the acceptance test passes.

Definitely open to a better way of handling this if you're aware of one - but in the meantime Patryk and I discussed eventually add Logpush auth resources separately and figuring out how to wire the http bucket based authentication scheme into an output of that new resource which the Logpush resource could take as an input.